test/test-parse.cpp

```diff
@@ -539,6 +540,121 @@ TEST(test_escape_first_character_in_keyword,
+             repeated_str(u8"[", u8"10", u8"]", parser::stack_limit - 2),
+             repeated_str(u8"{", u8"10", u8"}", parser::stack_limit - 2),
+             repeated_str(u8"while(true) ", u8"10", u8"",
+                          parser::stack_limit - 2),
+             repeated_str(u8"for(;;) ", u8"10", u8"", parser::stack_limit - 2),
+             repeated_str(u8"await ", u8"", u8"", parser::stack_limit - 2),
+             repeated_str(u8"if(true) ", u8"10", u8"", parser::stack_limit - 2),
+             repeated_str(u8"function f() { ", u8"", u8"}",
+                          parser::stack_limit - 1),
+             repeated_str(u8"() => { ", u8"", u8"}",
+                          (parser::stack_limit / 2) - 1),
+             repeated_str(u8"if(true) { ", u8"", u8"}",
+                          (parser::stack_limit / 2) - 1),
+             repeated_str(u8"while(true) { ", u8"", u8"}",
+                          (parser::stack_limit / 2) - 1),
+             repeated_str(u8"for(;;) { ", u8"", u8"}",
+                          (parser::stack_limit / 2) - 1),
+             repeated_str(u8"with({}) { ", u8"", u8"}",
+                          (parser::stack_limit / 2) - 1),
+             repeated_str(u8"do{ ", u8"", u8"} while (true);",
+                          (parser::stack_limit / 2) - 1),
+             repeated_str(u8"try{ ", u8"", u8"} catch(e) {}",
+                          parser::stack_limit - 1),
+             repeated_str(u8"class C { m() { ", u8"", u8"} }",
+                          parser::stack_limit - 1),
+         }) {
+      padded_string code(exps);
+      spy_visitor v;
+      parser p(&code, &v);
+      bool ok = p.parse_and_visit_module_catching_fatal_parse_errors(v);
+      EXPECT_TRUE(ok);
+      EXPECT_THAT(v.errors, IsEmpty());
+    }
+  }
+
+  {
+    padded_string code(
+        u8"(" + repeated_str(u8"{x:", u8"", u8"}", parser::stack_limit - 3) +
+        u8")");
+    spy_visitor v;
+    parser p(&code, &v);
+    bool ok = p.parse_and_visit_module_catching_fatal_parse_errors(v);
+    EXPECT_TRUE(ok);
+    EXPECT_THAT(v.errors, IsEmpty());
+  }
+}
+
+#if QLJS_HAVE_SETJMP
+TEST(test_overflow, parser_depth_limit_exceeded) {
+  for (const string8 &exps : {
+           repeated_str(u8"(", u8"10", u8")", parser::stack_limit + 1),
+           repeated_str(u8"[", u8"10", u8"]", parser::stack_limit + 1),
+           repeated_str(u8"{", u8"10", u8"}", parser::stack_limit + 1),
+           repeated_str(u8"while(true) ", u8"10", u8"",
+                        parser::stack_limit + 1),
+           repeated_str(u8"for(;;) ", u8"10", u8"", parser::stack_limit + 1),
+           repeated_str(u8"await ", u8"", u8"", parser::stack_limit + 1),
+           repeated_str(u8"if(true) ", u8"10", u8"", parser::stack_limit + 1),
+           repeated_str(u8"function f() { ", u8"", u8"}",
+                        parser::stack_limit + 1),
+           repeated_str(u8"() => { ", u8"", u8"}", parser::stack_limit / 2),
```

Comment:
Nit: I'd rather not be so precise.
```suggestion
           repeated_str(u8"() => { ", u8"", u8"}", parser::stack_limit + 1),
```